crates/ruff_python_formatter/src/pattern/pattern_match_sequence.rs

```diff
@@ -51,3 +43,54 @@ impl FormatNodeRule<PatternMatchSequence> for FormatPatternMatchSequence {
         }
     }
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(crate) enum SequenceType {
+    /// A list literal, e.g., `[1, 2, 3]`.
+    List,
+    /// A parenthesized tuple literal, e.g., `(1, 2, 3)`.
+    Tuple,
+    /// A tuple literal without parentheses, e.g., `1, 2, 3`.
+    TupleNoParens,
+}
+
+impl SequenceType {
+    pub(crate) fn from_pattern(pattern: &PatternMatchSequence, source: &str) -> SequenceType {
+        if source[pattern.range()].starts_with('[') {
+            SequenceType::List
+        } else if source[pattern.range()].starts_with('(') {
+            let Some(elt) = pattern.patterns.first() else {
```

Comment:
This seems to be the same as `is_tuple_parenthesized`. It may make sense to make the code reusable (as it is somewhat tricky). It would also be helpful to add an example of why the complex `open > close` paren path is necessary. I already forgot and had to look it up on the PR https://github.com/astral-sh/ruff/commit/95f78821adff8ee8b8d35158309c070bc2a00dfe

Reply:
Haha ok. I can try. It felt difficult to DRY it up without making the types much more permissive.

Reply:
Yeah, don't try too hard.